streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -137,45 +130,43 @@
     void to(Serde<K> keySerde, Serde<V> valSerde, String topic);
 
     /**
-     * Applies a stateful transformation to all elements in this stream.
+     * Create a new {@link KStream} instance by applying a stateful transformation to all elements in this stream.
      *
-     * @param transformerSupplier the class of valueTransformerSupplier
-     * @param stateStoreNames the names of the state store used by the processor
-     * @return the instance of KStream that contains transformed keys and values
+     * @param transformerSupplier   the instance of {@link TransformerSupplier} that generates stateful processors
+     * @param stateStoreNames       the names of the state store used by the processor
      */
     <K1, V1> KStream<K1, V1> transform(TransformerSupplier<K, V, KeyValue<K1, V1>> transformerSupplier, String... stateStoreNames);
 
     /**
-     * Applies a stateful transformation to all values in this stream.
+     * Create a new {@link KStream} instance by applying a stateful transformation to all values in this stream.
      *
-     * @param valueTransformerSupplier the class of valueTransformerSupplier
-     * @param stateStoreNames the names of the state store used by the processor
-     * @return the instance of KStream that contains the keys and transformed values
+     * @param valueTransformerSupplier  the instance of {@link ValueTransformerSupplier} that generates stateful processors
+     * @param stateStoreNames           the names of the state store used by the processor
      */
     <R> KStream<K, R> transformValues(ValueTransformerSupplier<V, R> valueTransformerSupplier, String... stateStoreNames);
 
     /**
-     * Processes all elements in this stream by applying a processor.
+     * Process all elements in this stream by applying a {@link org.apache.kafka.streams.processor.Processor}.
      *
-     * @param processorSupplier the supplier of the Processor to use
-     * @param stateStoreNames the names of the state store used by the processor
+     * @param processorSupplier         the supplier of {@link ProcessorSupplier} that generates processors
+     * @param stateStoreNames           the names of the state store used by the processor
      */
     void process(ProcessorSupplier<K, V> processorSupplier, String... stateStoreNames);
 
     /**
-     * Combines values of this stream with another KStream using Windowed Inner Join.
+     * Combine values of this stream with another {@link KStream} using windowed Inner Join.
```

Comment:
"Combine values of this stream [...]":  I'd clarify the role of the key(s) with regards to the two streams that are being joined.